tests/integration/test_kms.py

```diff
@@ -239,6 +240,48 @@ def test_generate_data_key_pair(self, kms_client, kms_key):
         )
         assert decrypted["Plaintext"] == result["PrivateKeyPlaintext"]
 
+    @pytest.mark.parametrize("number_of_bytes", [12, 44, 91, 1, 1024])
+    @pytest.mark.aws_validated
+    def test_generate_random(self, kms_client, snapshot, number_of_bytes):
+        result = kms_client.generate_random(NumberOfBytes=number_of_bytes)
+
+        plain_text = result.get("Plaintext")
+        assert plain_text
+        assert isinstance(plain_text, bytes)
+        assert len(plain_text) == number_of_bytes
+        snapshot.match("result_length", len(plain_text))
+
+    @pytest.mark.parametrize(
+        "number_of_bytes,error_type,error_message",
+        [
+            (
+                None,
+                botocore.exceptions.ClientError,
+                "NumberOfBytes is required.",
+            ),
+            (
+                0,
+                botocore.exceptions.ClientError,
+                "Member must have value greater than or equal to 1",
+            ),
+            (
+                1025,
+                botocore.exceptions.ClientError,
+                "Member must have value less than or equal to 1024",
+            ),
+        ],
+    )
+    @pytest.mark.aws_validated
+    def test_generate_random_invalid_number_of_bytes(
+        self, create_boto_client, snapshot, number_of_bytes, error_type, error_message
+    ):
+        kms_client = create_boto_client("kms", additional_config=Config(parameter_validation=False))
+        with pytest.raises(error_type) as exinfo:
+            kms_client.generate_random(NumberOfBytes=number_of_bytes)
+
+        assert error_message in str(exinfo.value)
+        snapshot.match("error_message", str(exinfo.value))
```

Comment:
We usually snapshot the whole response. Here, this would be `snapshot.match("generate-random-exc", e.value.response)`. This leads to the response by AWS/localstack being snapshotted, not just the message. It could have some additional attributes, for some services.
The assert of the error message will also be taken care of by the snapshot, as mentioned above.